js/modules/k6/k6.go

```diff
@@ -160,10 +161,11 @@ func (*K6) Check(ctx context.Context, arg0, checks goja.Value, extras ...goja.Va
 		fn, ok := goja.AssertFunction(val)
 		if ok {
 			tmpVal, err := fn(goja.Undefined(), arg0)
+			val = tmpVal
 			if err != nil {
-				return false, err
+				val = rt.ToValue(false)
+				exc = err
```

Comment:
Hmm I'm not sure this is the correct behavior we're looking for. For one, it is actually a minor breaking change... Currently, as soon as one of the callbacks throws an error, the execution will be aborted and the other callbacks won't be called. This changes that - even if a callback throws an error, that error will be suppressed until the other callbacks are also executed. What's worse, if one of the other callbacks also throws an error, that error will overwrite the first error.

Reply:
I'm not sure you're right, but correct me if I'm wrong... :)

Take a look at the [changed test](https://github.com/loadimpact/k6/blob/af0b23d320f98ed544415d33a2953942e7803914/js/modules/k6/k6_test.go#L248-L273). I only added the metrics check to ensure the failure is correctly emmitted (which fails without this change), but the assertion is still that it returns `Error: error A at a (<eval>:3:27(6))`, i.e. the first error seen in the order the checks are executed, so this doesn't change that.

Reply:
hmm I see the test, but I'm still not convinced... I don't know how goja iterates over maps and if the order of the `Object.Keys()` result is constant, so the test passing might just be a fluke. Looking at the new code here, it seems subtly wrong to me, but I might be wrong :confused:

Reply:
ah, sorry, my mistake - didn't see that the `return` is still in the `for` loop, everything is fine :smile: